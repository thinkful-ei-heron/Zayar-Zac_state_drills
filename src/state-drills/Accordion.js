import React from 'react';

export default class Accordion extends React.Component {

	static defaultProps = {
		sections: [{}]
	}

    constructor(props) {
        super(props);
        this.state = {
            openIndex: null
        }
    }

	btnHandler = index => {
		this.setState({
			openIndex: index
		});
	}

    render() {
        
        let itemButtons = this.props.sections.map((section, index) => {
            return (
                <li key={index}>
                    <button className={'btnTest'} key={index} onClick={() => this.btnHandler(index)}>{section.title}</button>
					{(this.state.openIndex === index) && <p>{section.content}</p>}
				</li>
            )
        })

		return (
            <ul>
                <li>test</li>
                {itemButtons}
			</ul>

		);
	};
}