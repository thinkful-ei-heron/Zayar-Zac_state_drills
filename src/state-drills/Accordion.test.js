import React from 'react';
import ReactDOM from 'react-dom'
import Accordion from './Accordion';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Accordion component', () => {
	const sections = [
		{
		  title: 'Section 1',
		  content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
		},
		{
		  title: 'Section 2',
		  content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
		},
		{
		  title: 'Section 3',
		  content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
		},
	  ]

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
	})
	
	it('renders an empty li when the sections prop is not supplied', () => {
        const wrapper = shallow(<Accordion />)
    	expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders no sections as active by default', () => {
		const wrapper = shallow(<Accordion section={sections} />)
    	expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('opens a clicked section', () => {
		const wrapper = shallow(<Accordion section={sections} />)
		wrapper.find('button').at(0).simulate('click')
    	expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('only opens the last section when multiple sections have been clicked', () => {
        const wrapper = shallow(<Accordion section={sections} />)
        console.log(wrapper.debug());
        wrapper.find('.btn-Test').at(0).simulate('click')
        console.log(wrapper.debug());
		wrapper.find('.btn-Test').at(1).simulate('click')
    	expect(toJson(wrapper)).toMatchSnapshot()
	})
})