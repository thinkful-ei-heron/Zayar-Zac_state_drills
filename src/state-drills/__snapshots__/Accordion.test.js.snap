// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    />
    <p />
  </li>
</ul>
`;

exports[`Accordion component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      className="btnTest"
      key="0"
      onClick={[Function]}
    />
    <p />
  </li>
</ul>
`;

exports[`Accordion component renders an empty li when the sections prop is not supplied 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      className="btnTest"
      key="0"
      onClick={[Function]}
    />
  </li>
</ul>
`;

exports[`Accordion component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      className="btnTest"
      key="0"
      onClick={[Function]}
    />
  </li>
</ul>
`;
